EUR-Lex.js

```diff
@@ -105,112 +115,175 @@ function doWeb(doc, url) {
-				item.code = uriParts[0];
-				item.codeNumber = uriParts[1] + ', ' + uriParts[2];
-			} else {
-				item.code = uriParts[0] + ' ' + uriParts[1];
-				item.codeNumber = uriParts[3];
-			}
-			if (type=="bill") {
-				item.codeVolume = item.code;
-				item.code = item.codeNumber;
+
+	if (eliTypeUri) {
+		// type: everything with ELI (see var typeMapping: bill, statute, report)
+		item.title = attr(doc, 'meta[property="eli:title"][lang=' + languageUrl.toLowerCase() + "]", "content");
+		var uri = attr(doc, "#format_language_table_digital_sign_act_" + languageUrl.toUpperCase(), "href");
+		if (uri) {
+			var uriParts = uri.split("/").pop().replace("?uri=", "").split(":");
+			// e.g. uriParts =  ["OJ", "L", "1995", "281", "TOC"]
+			// e.g. uriParts = ["DD", "03", "061", "TOC", "FI"]
+			if (uriParts.length >= 4) {
+				if (/\d+/.test(uriParts[1])) {
+					item.code = uriParts[0];
+					item.codeNumber = uriParts[1] + ", " + uriParts[2];
+				} 
+				else {
+					item.code = uriParts[0] + " " + uriParts[1];
+					item.codeNumber = uriParts[3];
+				}
+				if (type == "bill") {
+					item.codeVolume = item.codeNumber;
+					item.codeNumber = null;
+				}
 			}
 		}
+
+		// item.number = attr(doc, 'meta[property="eli:id_local"]', 'content');
+
+		item.date = attr(doc, 'meta[property="eli:date_document"]', "content");
+		if (!item.date) {item.date = attr(doc, 'meta[property="eli:date_publication"]', "content")}
+		
+		var passedBy = doc.querySelectorAll('meta[property="eli:passed_by"]');
+		var passedByArray = [];
+		for (let i = 0; i < passedBy.length; i++) {
+			passedByArray.push(passedBy[i].getAttribute("resource").split("/").pop());
+		}
+		item.legislativeBody = passedByArray.join(", ");
+		
+		item.url = attr(doc, 'meta[typeOf="eli:LegalResource"]', "about") + "/" + language.toLowerCase();
+		// eli:is_about -> eurovoc -> tags
 	}
-	
-	item.number = attr(doc, 'meta[property="eli:id_local"]', 'content');
-	
-	item.date = attr(doc, 'meta[property="eli:date_publication"]', 'content');
-	// attr(doc, 'meta[property="eli:date_document"]', 'content');
+	else if (docSector == "6") {
+		// type: case
+		// pretty hacky stuff, as there's little metadata available
+		var docCourt = docType.substr(0, 1);
+		if (docCourt == "C") {
+			item.court = languageMapping[languageUrl][1] || languageMapping["EN"][1];
+		}
+		else if (docCourt == "T") {
+			item.court = languageMapping[languageUrl][2] || languageMapping["EN"][2];
+		}
+		else if (docCourt == "F") {
+			item.court = languageMapping[languageUrl][3] || languageMapping["EN"][3];
+		}
+		item.url = url;
 
-	var passedBy = doc.querySelectorAll('meta[property="eli:passed_by"]');
-	var passedByArray = [];
-	for (let i=0; i<passedBy.length; i++) {
-		passedByArray.push(passedBy[i].getAttribute('resource').split('/').pop());
+		if (docType.substr(1) == "J") { // Judgments
+			var titleParts = attr(doc, 'meta[name="WT.z_docTitle"]', "content").replace(/\./g,"").split("#");
+			item.caseName = titleParts[1];
```

Comment:
Add a check to make sure the `split` returned the right number of segments first - this is pretty fragile.